airflow/providers/oracle/operators/oracle.py

```diff
@@ -47,19 +49,26 @@ class OracleOperator(BaseOperator):
     def __init__(
         self,
         *,
-        sql: Union[str, List[str]],
+        sql: Optional[Union[str, List[str]]] = None,
+        procedure: Optional[str] = None,
         oracle_conn_id: str = 'oracle_default',
         parameters: Optional[Union[Mapping, Iterable]] = None,
         autocommit: bool = False,
         **kwargs,
     ) -> None:
+        if sql is None and procedure is None:
+            raise ValueError("Must provide either 'sql' or 'procedure'")
         super().__init__(**kwargs)
         self.oracle_conn_id = oracle_conn_id
         self.sql = sql
+        self.procedure = procedure
         self.autocommit = autocommit
         self.parameters = parameters
 
     def execute(self, context) -> None:
-        self.log.info('Executing: %s', self.sql)
+        self.log.info('Executing: %s', self.sql or self.procedure)
         hook = OracleHook(oracle_conn_id=self.oracle_conn_id)
-        hook.run(self.sql, autocommit=self.autocommit, parameters=self.parameters)
+        if self.sql:
```

Comment:
I think these should be two separate operators- one for running SQL statements and one for calling procedures.

Reply:
@mik-laj it does perhaps make sense considering that autocommit (see also #20085) is disabled in the operator that runs SQL statements while for calling stored procedures – in the context of an Airflow operator – it makes sense to autocommit (given that the stored procedure itself runs inside a subtransaction).

Reply:
@mik-laj I have split the functionality now, introducing a new operator `OracleStoredProcedureOperator`. This always has `autocommit` enabled – because that's really what makes sense.